### PR TITLE
Stats Widgets: clear stored data when site is changed

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref #12702 

When changing the site used for widgets, the stored data for the old site is now purged.

To test:
The easiest way to test this is to prevent the widgets from fetching data for the new site.

- In the app, go to Stats > Widgets > `Use this site`.
- Add the 3 WordPress widgets to the Today view.
- Return to the app and change the site used for widgets.
- Disable your internet connection.
- Return to the Today view.
- Verify all the widgets show dashes (because no data was loaded from the plists and new data cannot be fetched).

**Note:** I'm currently working on a "no connection" view, so showing dashes in this state is temporary. This is just the easiest way to verify the data is purged.

no_data

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Today Widgets/Data/AllTimeWidgetStats.swift [hunk 1]
```diff
@@ -34,7 +34,20 @@ extension AllTimeWidgetStats {
...
         } catch {
             DDLogError("Failed loading AllTimeWidgetStats data: \(error.localizedDescription)")
-            return AllTimeWidgetStats()
+            return nil
+        }
+    }
+
+    static func clearSavedData() {
+        guard let dataFileURL = AllTimeWidgetStats.dataFileURL else {
+            return
```
Comment (bjtitus) [inline WordPress/Classes/ViewRelated/Stats/Today Widgets/Data/AllTimeWidgetStats.swift]: Is it worth adding any kind of logging here to track when the file URL was empty?

File: WordPress/Classes/ViewRelated/Stats/Today Widgets/Data/ThisWeekWidgetStats.swift [hunk 2]
```diff
@@ -45,7 +45,20 @@ extension ThisWeekWidgetStats {
...
         } catch {
             DDLogError("Failed loading ThisWeekWidgetStats data: \(error.localizedDescription)")
-            return ThisWeekWidgetStats()
+            return nil
+        }
+    }
+
+    static func clearSavedData() {
+        guard let dataFileURL = ThisWeekWidgetStats.dataFileURL else {
+            return
```
Comment (bjtitus) [inline WordPress/Classes/ViewRelated/Stats/Today Widgets/Data/ThisWeekWidgetStats.swift]: Is it worth logging this so we have some idea when this URL is empty?

File: WordPress/Classes/ViewRelated/Stats/Today Widgets/Data/TodayWidgetStats.swift [hunk 3]
```diff
@@ -21,20 +21,33 @@ struct TodayWidgetStats: Codable {
...
-            DDLogError("Failed loading TodayWidgetStats data: \(error.localizedDescription)")
-            return TodayWidgetStats()
+            DDLogError("TodayWidgetStats: Failed loading data: \(error.localizedDescription)")
+            return nil
+        }
+    }
+
+    static func clearSavedData() {
+        guard let dataFileURL = TodayWidgetStats.dataFileURL else {
+            return
```
Comment (bjtitus) [inline WordPress/Classes/ViewRelated/Stats/Today Widgets/Data/TodayWidgetStats.swift]: Is it worth logging here so we know when this URL is missing?

**Comment by ScoutHarris** [comment]
Hey @bjtitus .

> Is it worth logging here so we know when this URL is missing?

Sure, I can do that. I'll add those in my next PR to catch all the places that check the `dataFileURL`. Thanks!

**Pull request merged**